hugolib/author.go

```diff
@@ -13,23 +13,51 @@
 
 package hugolib
 
+import (
+	"fmt"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/spf13/cast"
+)
+
+var (
+	onlyNumbersRegExp = regexp.MustCompile("^[0-9]*$")
+)
+
 // AuthorList is a list of all authors and their metadata.
 type AuthorList map[string]Author
 
+// Authors returns a slice of authors sorted by ID
+func (al AuthorList) Authors() []Author {
```

Comment:
I can remove some of the sorting methods, maybe the Desc ones, but I think Authors are fundamentally different than other sorting options in Hugo. A menu or a taxonomy doesn't really have a logical way to be ordered, but it's very logical to order a list of people sorted by their names. I fully expect people to add `/authors/` pages that list all the authors on the site.

I agree that we don't want people changing their IDs, so weight is a good option, but by that same token, people aren't going to want to go in and change the weight of every single person each time they add an author to make sure that names sort alphabetically.